Parsers/ASimWebSession/Parsers/ASimWebSessionSonicWallFirewall.yaml

```diff
@@ -0,0 +1,401 @@
+Parser:
+  Title: Web Session ASIM parser for SonicWall firewalls
+  Version: '0.1.0'
+  LastUpdated: Feb 28, 2024
+Product:
+  Name: SonicWall
+Normalization:
+  Schema: WebSession
+  Version: '0.2.6'
+References:
+  - Title: ASIM Web Session Schema
+    Link: https://aka.ms/ASimWebSessionDoc
+  - Title: ASIM
+    Link: https:/aka.ms/AboutASIM
+  - Title: SonicWall SonicOS/X 7.0.1 Log Events Reference Guide [PDF]
+    Link: https://www.sonicwall.com/techdocs/pdf/SonicOS-X_7.0.1_LogEvents_ReferenceGuide.pdf
+Description: |
+  This ASIM parser supports normalizing SonicWall SonicOS ArcSight-formatted syslog ingested by Microsoft Sentinel to the ASIM WebSession normalized schema.
+ParserName: ASimWebSessionSonicWallFirewall
+EquivalentBuiltInParser: _ASim_WebSession_SonicWallFirewall
+ParserParams:
+  - Name: disabled
+    Type: bool
+    Default: false
+ParserQuery: |
+  let parser=(disabled:bool=false){
+      let Actions=datatable(fw_action:string, DvcAction:string, EventSeverity:string)
+        [ "\"forward\"", "Allow", "Informational"
+        , "\"mgmt\"", "Other", "Informational"
+        , "\"NA\"", "Other", "Informational"
+        , "\"drop\"", "Drop", "Low"
+      ];
+      CommonSecurityLog
+      | where not(disabled)
+      | where DeviceVendor == "SonicWall"
+      | where DeviceEventClassID in (14, 97)
+      | where Protocol has_any(dynamic(["udp/http", "tcp/http", "udp/https", "tcp/https"]))
+      | parse-kv AdditionalExtensions as (['gcat']:string, ['app']:string, ['arg']:string, ['dstV6']:string, ['srcV6']:string, ['snpt']:string, ['dnpt']:string, ['susr']:string,['appName']:string, ['appcat']:string, ['appid']:string, ['sid']:string, ['catid']:string, ['ipscat']:string, ['ipspri']:string, ['spycat']:string, ['spypri']:string, ['fw_action']:string, ['dpi']:string, ['bid']:string, ['af_action']:string, ['af_polid']:string, ['af_policy']:string, ['af_type']:string, ['af_service']:string, ['af_object']:string, ['contentObject']:string, ['fileid']:string, ['uuid']:string) with (pair_delimiter=";", kv_delimiter="=")
+      | extend

```

Comment:
Are you saying instead of extending to SourceIP, extend to SrcIpAddr? I'm not sure why that would be a problem... it gets renamed to the SrcIpAddr a bit later in the file.

How about this? Is this better?
`      | extend
        SrcIpAddr = coalesce(SourceIP, srcV6)
        , DstIpAddr = coalesce(DestinationIP, dstV6)
      | where (isnotempty(SrcIpAddr) or isnotempty(DstIpAddr))`

Reply:
@jaimeesc not a problem, just helps with reducing additional steps. You don't have to rename later if actual field used in the first place.